db/write_batch.cc

```diff
@@ -1746,6 +1746,30 @@ class MemTableInserter : public WriteBatch::Handler {
   }
 };
 
+class MemTableInserterNoFlush : public MemTableInserter {
```

Comment:
After some debate, it seems that we can remove this MemTableInserterNoFlush class and change the `CheckMemtableFull` method to non virtual.